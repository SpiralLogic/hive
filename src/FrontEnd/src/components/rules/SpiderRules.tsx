import { FunctionComponent, h } from 'preact';
import Hexagon from '../Hexagon';
import Hextille from '../Hextille';
import Row from '../Row';
import RuleCell from './RuleCell';

const SpiderRules: FunctionComponent = () => {
  return (
    <Hextille class="rules">
      <Row>
        <RuleCell result="correct" />
        <RuleCell result="correct" symbol="2" correctArrows={['left']} />
        <RuleCell class="selected" zIndex={4} creature="spider" correctArrows={['right', 'bottomLeft']} />
        <RuleCell result="correct" zIndex={3} symbol="1" correctArrows={['bottomRight']} />
      </Row>
      <Row>
        <RuleCell creature="beetle" />
        <RuleCell
          result="correct"
          symbol="1"
          zIndex={3}
          correctArrows={['topLeft', 'bottomLeft', 'bottomRight']}
        />
        <RuleCell creature="spider" />
        <RuleCell result="correct" zIndex={2} symbol="2" correctArrows={['bottomRight']} />
      </Row>
      <Row>
        <RuleCell creature="ant" />
        <RuleCell result="correct" zIndex={1} symbol="2" correctArrows={['bottomRight']} />
        <RuleCell result="correct" zIndex={1} symbol="2" correctArrows={['bottomLeft', 'bottomRight']} />
        <RuleCell creature="queen" />
        <RuleCell result="correct" />
      </Row>
      <Row>
        <RuleCell creature="queen" />
        <RuleCell result="correct" />
        <RuleCell result="correct" />
        <RuleCell creature="beetle" />
        <Hexagon hidden={true} />
      </Row>
      <Row>
        <RuleCell creature="grasshopper" />
        <RuleCell creature="grasshopper" />
        <RuleCell creature="ant" />
      </Row>
    </Hextille>
  );
};
SpiderRules.displayName = 'SpiderRules';
export default SpiderRules;
