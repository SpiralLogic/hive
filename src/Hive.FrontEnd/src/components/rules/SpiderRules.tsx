import Hexagon from '../Hexagon';
import Hextille from '../Hextille';
import Row from '../Row';
import RuleCell from './RuleCell';
import { Rule } from './rule';

const SpiderRules: Rule = () => (
  <Hextille class="rules">
    <Row>
      <Hexagon hidden={true} />
      <RuleCell creature="grasshopper" />
      <RuleCell creature="grasshopper" />
      <RuleCell creature="ant" />
    </Row>
    <Row>
      <Hexagon hidden={true} />
      <RuleCell creature="queen" />
      <RuleCell result="correct" />
      <RuleCell result="correct" />
      <RuleCell creature="beetle" />
    </Row>{' '}
    <Row>
      <RuleCell creature="ant" />
      <RuleCell result="correct" zIndex={1} symbol="2" correctArrows={['topRight']} />
      <RuleCell result="correct" zIndex={1} symbol="2" correctArrows={['topLeft', 'topRight']} />
      <RuleCell creature="queen" />
      <RuleCell result="correct" />
    </Row>
    <Row>
      <Hexagon hidden={true} />
      <RuleCell creature="beetle" />
      <RuleCell
        result="correct"
        symbol="1"
        zIndex={3}
        correctArrows={['topLeft', 'bottomLeft', 'topRight']}
      />
      <RuleCell creature="spider" />
      <RuleCell result="correct" zIndex={2} symbol="2" correctArrows={['topRight']} />
    </Row>{' '}
    <Row>
      <RuleCell result="correct" />
      <RuleCell result="correct" symbol="2" correctArrows={['left']} />
      <RuleCell selected creature="spider" correctArrows={['right', 'topLeft']} />
      <RuleCell result="correct" zIndex={3} symbol="1" correctArrows={['topRight']} />
    </Row>
  </Hextille>
);

SpiderRules.title = 'Spider';
SpiderRules.description = ['Must move 3 cells', `Can't backtrack`];
SpiderRules.displayName = 'SpiderRules';

export default SpiderRules;
