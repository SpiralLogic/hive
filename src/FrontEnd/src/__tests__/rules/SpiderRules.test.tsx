import { h } from 'preact';
import { renderElement } from '../test-helpers';
import SpiderRules from '../../components/rules/SpiderRules';

describe('Spider Rule snapshot tests', () => {
  test('snapshot', () => {
    expect(renderElement(<SpiderRules />)).toMatchSnapshot();
  });
});
