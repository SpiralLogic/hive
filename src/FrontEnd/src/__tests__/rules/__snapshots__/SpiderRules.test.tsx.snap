// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spider Rule snapshot tests snapshot 1`] = `
<div
  class="left"
  role="grid"
>
  <div
    role="row"
  >
    <div
      class="correct"
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
        </svg>
        <span>
          ✓
        </span>
      </div>
    </div>
    <div
      class="correct"
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
        </svg>
        <span>
          2
        </span>
      </div>
      <svg
        class="arrow left correct"
        style="--span: 1;"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#arrow"
        />
      </svg>
    </div>
    <div
      role="cell"
      style="z-index: 4;"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile spider selected"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#spider"
          />
        </svg>
      </div>
      <svg
        class="arrow right correct"
        style="--span: 1;"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#arrow"
        />
      </svg>
      <svg
        class="arrow bottomLeft correct"
        style="--span: 1;"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#arrow"
        />
      </svg>
    </div>
    <div
      class="correct"
      role="cell"
      style="z-index: 3;"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
        </svg>
        <span>
          1
        </span>
      </div>
      <svg
        class="arrow bottomRight correct"
        style="--span: 1;"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#arrow"
        />
      </svg>
    </div>
  </div>
  <div
    role="row"
  >
    <div
      role="none"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
    </div>
    <div
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile beetle"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#beetle"
          />
        </svg>
      </div>
    </div>
    <div
      class="correct"
      role="cell"
      style="z-index: 3;"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
        </svg>
        <span>
          1
        </span>
      </div>
      <svg
        class="arrow topLeft correct"
        style="--span: 1;"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#arrow"
        />
      </svg>
      <svg
        class="arrow bottomLeft correct"
        style="--span: 1;"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#arrow"
        />
      </svg>
      <svg
        class="arrow bottomRight correct"
        style="--span: 1;"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#arrow"
        />
      </svg>
    </div>
    <div
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile spider"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#spider"
          />
        </svg>
      </div>
    </div>
    <div
      class="correct"
      role="cell"
      style="z-index: 2;"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
        </svg>
        <span>
          2
        </span>
      </div>
      <svg
        class="arrow bottomRight correct"
        style="--span: 1;"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#arrow"
        />
      </svg>
    </div>
  </div>
  <div
    role="row"
  >
    <div
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile ant"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#ant"
          />
        </svg>
      </div>
    </div>
    <div
      class="correct"
      role="cell"
      style="z-index: 1;"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
        </svg>
        <span>
          2
        </span>
      </div>
      <svg
        class="arrow bottomRight correct"
        style="--span: 1;"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#arrow"
        />
      </svg>
    </div>
    <div
      class="correct"
      role="cell"
      style="z-index: 1;"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
        </svg>
        <span>
          2
        </span>
      </div>
      <svg
        class="arrow bottomLeft correct"
        style="--span: 1;"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#arrow"
        />
      </svg>
      <svg
        class="arrow bottomRight correct"
        style="--span: 1;"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#arrow"
        />
      </svg>
    </div>
    <div
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile queen"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#queen"
          />
        </svg>
      </div>
    </div>
    <div
      class="correct"
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
        </svg>
        <span>
          ✓
        </span>
      </div>
    </div>
  </div>
  <div
    role="row"
  >
    <div
      role="none"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
    </div>
    <div
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile queen"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#queen"
          />
        </svg>
      </div>
    </div>
    <div
      class="correct"
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
        </svg>
        <span>
          ✓
        </span>
      </div>
    </div>
    <div
      class="correct"
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
        </svg>
        <span>
          ✓
        </span>
      </div>
    </div>
    <div
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile beetle"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#beetle"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    role="row"
  >
    <div
      role="none"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
    </div>
    <div
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile grasshopper"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#grasshopper"
          />
        </svg>
      </div>
    </div>
    <div
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile grasshopper"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#grasshopper"
          />
        </svg>
      </div>
    </div>
    <div
      role="cell"
    >
      <svg
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          href="#hex"
        />
      </svg>
      <div
        class="tile ant"
      >
        <svg
          viewBox="0 0 100 100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            href="#hex"
          />
          <use
            class="creature"
            href="#ant"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
